examples/server.rs

```diff
@@ -0,0 +1,159 @@
+use h3_quinn::quinn::{Certificate, CertificateChain, PrivateKey};
+use std::path::PathBuf;
+use structopt::StructOpt;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+use tokio::stream::StreamExt;
+use tracing::{debug, error, trace, trace_span, warn};
+
+// Configs for two server modes
+// selfsigned mode will generate it's own local certificate
+// certs mode will require a path to 2-3 files(cert, key, ca)
+#[derive(StructOpt, Debug)]
+#[structopt(name = "server")]
+struct Opt {
+    #[structopt(subcommand)]
+    pub command: Command,
+}
+
+#[derive(StructOpt, Debug)]
+pub enum Command {
+    #[structopt(name = "selfsigned")]
```

Comment:
Very cool!

Reply:
I picked up this technique from the Tonari folks post here: https://blog.tonari.no/why-we-love-rust

It's made cli applications a TON easier.